src/lib/core/config.js

```diff
@@ -59,6 +59,11 @@ var Config = function(options) {
     };
     self.contractsFiles.push(new File({filename, type: File.types.custom, path: filename, resolver}));
   });
+
+  self.events.on('file-remove', (fileType, path) => {
+    if(fileType !== 'contract') return;
+    self.contractsFiles = self.contractsFiles.filter(file => file.filename !== path);
+  });
```

Comment:
There is currently no event listening on this file, I am wondering if this is the right place for this code.
This file seems to be responsible for: parsing contract files.

Maybe I am wrong but I would see this code more in pipeline.
On file-remove => reload contracts config.
For example we call this function `engine.config.reloadConfig();`

Probably @jrainville or @iurimatias can give a more accurate answer

Reply:
Not sure I understand your comment. You mean no one is listening for `contractFiles` changes? Because it's a reference, so any file doing `config.contractFiles` will get the right ones, especially with an event call.

Reply:
I mean if you open the file: src/lib/core/config.js
This file doesn't have any mention of: `events.on(something)`

If I think about responsibility: It is about parsing config file, by adding `events.on` we are adding to it a new responsibility.

Hence my questions:
- Is the config file responsible for file management (file-remove)
- Is this code should be here according to our current architecture.

That is why I suggest the pipeline
Better?

Reply:
But... it does have those, it's just that they are `commandHandler`s instead, which under the hood, are the same. 
eg: https://github.com/embark-framework/embark/blob/c14c1f34e443c4e6537964cbde45c15e9199c971/src/lib/core/config.js#L56

Reply:
I feel that if it remembers the contract files, and it responds to "add" it should react to a removal too.

In other words, I agree with @jrainville in that this feels like the responsible party for tracking contracts _EVEN THOUGH_ we have a contracts manager.

Reply:
Fair enough 👍